Bug 1458618 - Send periodic alerts when there is a blocker bug

helfi92 opened the pull request.

helfi92 committed:
Send periodic alerts when there is a blocker bug

eliperelman commented on README.md:
```diff
@@ -29,10 +32,18 @@ And the following are optional:
 to [hubot-irc#configuring-the-adapter](https://github.com/nandub/hubot-irc/blob/master/README.md#configuring-the-adapter).
 
-_Example: Configuring for mozilla's taskcluster-bots IRC channel_
+To get started local development, create a `.env` file in the root of the repo.
```
You don't *have* to create a `.env` file, these can be set in the environment as well.

eliperelman commented on test/api_test.js:
```diff
@@ -86,4 +86,17 @@ describe('Array', function() {
+    const BUGZILLA_BASE_URL = 'https://bugzilla.mozilla.org';
+    const FIRST_BUG_NUMBER = 35;
+
+    it('should fetch the first bug ever filed on bugzilla', async () => {
```
I'm not sure this test is useful, as this is testing the API, not any functionality of this tool.

eliperelman commented on scripts/index.js:
```diff
@@ -29,6 +31,10 @@
 //   helfi92
+const got = require('got');
+
+const BUGZILLA_BASE_URL = 'https://bugzilla.mozilla.org';
```
You can set this in the environment as well so it can be overridden. Create a `.env` file in this repo and force commit the file, but leave it in `.gitignore`.

eliperelman commented on scripts/index.js:
```diff
@@ -52,6 +58,29 @@ module.exports = (robot) => {
+      } catch (err) {
+        robot.logger.error(err);
+      }
+    }, process.env.BUGZILLA_BLOCKER_TIME_PERIOD || 5 * 60 * 1000);
```
If you create a default `.env` file, you can set this default value there:

```
BUGZILLA_BLOCKER_TIME_PERIOD=300000
```

```js
parseInt(process.env.BUGZILLA_BLOCKER_TIME_PERIOD, 10)
```

Maybe rename to `BUGZILLA_BLOCKER_INTERVAL`.

helfi92 replied:
Very good idea. I will revise.

helfi92 committed:
Update test and have better usage of env vars

Turns out a hubot script can live in a src/scripts directory.
This commit moves scritps/ to src/scripts as well.

helfi92 committed:
Force push a default .env file

helfi92 committed:
Nit

helfi92 committed:
Add exponential backoff interval check to blockers

eliperelman reviewed (approved):
Neat!
